test/common/encoding.h

```diff
@@ -0,0 +1,639 @@
+		//| dolar       |  cent        |    euro  
+		unsigned char utf32be[] = {0, 0, 0, 0x24, 0, 0, 0, 0xA2, 0, 0, 0x20, 0xAC, 0, 0, 0, 0};
+
+		unsigned char *utf16 = utf16be;
+		unsigned char *utf32 = utf32be;
+#else
+		//| dolar |  cent  |    euro   |
+		unsigned char utf16le[] = {0x24, 0, 0xA2, 0, 0xAC, 0x20, 0, 0};
+		//| dolar       |  cent        |    euro
+		unsigned char utf32le[] = {0x24, 0, 0, 0, 0xA2, 0, 0, 0, 0xAC, 0x20, 0, 0, 0, 0, 0, 0};
+
+		unsigned char *utf16 = utf16le;
+		unsigned char *utf32 = utf32le;
+#endif
+
+		// UTF16 to UTF8
+		Common::String resultstr8 = Common::U32String::decodeUTF16Native((uint16 *) utf16, 3).encode(Common::kUtf8);
+		TS_ASSERT(resultstr8.c_str() != NULL);
+		TS_ASSERT_EQUALS(memcmp(resultstr8.c_str(), utf8, 7), 0)
+
+		// UTF32 to UTF8
+
+		resultstr8 = Common::U32String((uint32 *) utf32, 3).encode(Common::kUtf8);
+		TS_ASSERT(resultstr8.c_str() != NULL);
+		TS_ASSERT_EQUALS(memcmp(resultstr8.c_str(), utf8, 7), 0);
+
+		// UTF32 to UTF16
+		uint16 *result16 = Common::U32String((uint32 *) utf32, 3).encodeUTF16Native(NULL);
+		TS_ASSERT(result16 != NULL);
+		TS_ASSERT_EQUALS(memcmp(result16, utf16, 8), 0);
+		free(result16);
+
+		// UTF8 to UTF16
+
+		result16 = Common::U32String((char *) utf8, 6, Common::kUtf8).encodeUTF16Native(NULL);
+		TS_ASSERT(result16 != NULL);
+		TS_ASSERT_EQUALS(memcmp(result16, utf16, 8), 0);
+		free(result16);
+
+		// UTF8 to UTF32
+		Common::U32String resultustr = Common::String((const char *) utf8, 6).decode(Common::kUtf8);
+		TS_ASSERT_EQUALS(memcmp(resultustr.c_str(), utf32, 16), 0);
+
+		// UTF16 to UTF32
+		resultustr = Common::U32String::decodeUTF16Native((uint16 *) utf16, 3);
+		TS_ASSERT_EQUALS(memcmp(resultustr.c_str(), utf32, 16), 0);
+	}
+
+	void test_conversion_unicode_big_endian() {
+		Common::install_mock_g_system();
+		//  |dolar|   cent    |     euro       |
+		unsigned char utf8[] = {0x24, 0xC2, 0xA2, 0xE2, 0x82, 0xAC, 0};
+		//| dolar |  cent  |    euro   |
+		unsigned char utf16be[] = {0, 0x24, 0, 0xA2, 0x20, 0xAC, 0, 0};
+		//| dolar       |  cent        |    euro  
+		unsigned char utf32be[] = {0, 0, 0, 0x24, 0, 0, 0, 0xA2, 0, 0, 0x20, 0xAC, 0, 0, 0, 0};
+
+#if SCUMM_BIG_ENDIAN
+		//| dolar |  cent  |    euro   |
+		unsigned char utf16[] = {0, 0x24, 0, 0xA2, 0x20, 0xAC, 0, 0};
+		//| dolar       |  cent        |    euro
+		unsigned char utf32[] = {0, 0, 0, 0x24, 0, 0, 0, 0xA2, 0, 0, 0x20, 0xAC, 0, 0, 0, 0};
+#else
+		//| dolar |  cent  |    euro   |
+		unsigned char utf16[] = {0x24, 0, 0xA2, 0, 0xAC, 0x20, 0, 0};
+		//| dolar       |  cent        |    euro
+		unsigned char utf32[] = {0x24, 0, 0, 0, 0xA2, 0, 0, 0, 0xAC, 0x20, 0, 0, 0, 0, 0, 0};
+#endif
+
+		// UTF16 to UTF8
+		Common::String resultstr8 = Common::U32String::decodeUTF16BE((uint16 *) utf16be, 3).encode(Common::kUtf8);
+		TS_ASSERT(resultstr8.c_str() != NULL);
+		TS_ASSERT_EQUALS(memcmp(resultstr8.c_str(), utf8, 7), 0);
+
+#if 0
```

Comment:
Maybe all those tests with UTF-32LE and UTF-32BE could be removed altogether? Or is there a good raison to keep the code? As far as I saw only UTF-32 in native endianness is supported now, and I don't see that we will need to add back support for specific endianness anytime soon (and endianness can be handled separately from string encoding conversion).